lib/node_modules/@stdlib/_tools/changelog/generate/lib/main.js

```diff
@@ -0,0 +1,388 @@
+	var files;
+	var bool;
+	var i;
+
+	files = commit.files;
+	bool = false;
+	for ( i = 0; i < files.length; i++ ) {
+		if ( files[ i ].indexOf( 'lib/node_modules/@stdlib/_tools/' ) === -1 ) {
+			bool = true;
+			break;
+		}
+	}
+	return bool;
+}
+
+/**
+* Wraps a package summary for inclusion in a namespace changelog.
+*
+* @private
+* @param {string} pkg - namespace package
+* @param {string} version - namespace version
+* @param {string} name - subpackage name
+* @param {string} summary - package summary
+* @returns {string} wrapped package summary
+*/
+function packageSummaryWrapper( pkg, version, name, summary ) {
+	var href;
+	var out;
+	var id;
+
+	name = pkg + ( ( name ) ? '/' + name : '' );
+	href = STDLIB_REPO_NODE_PATH + '/%40stdlib/' + name;
+	id = replace( name, '/', '-' );
+	if ( version ) {
+		id += '-v' + version;
+	} else {
+		id += '-unreleased';
+	}
+	out = '<section class="package" id="' + id + '">\n\n';
+	out += heading( '[@stdlib/' + name + '](' + href + ')', 4 );
+	out += '<details>\n\n';
+	out += decrementHeaders( summary, 2 );
+	out += '</details>';
+	out += sectionEnd( 'package' );
+	return out;
+}
+
+
+// MAIN //
+
+/**
+* Generates a Markdown formatted changelog for a specified package.
+*
+* @param {string} pkg - package name
+* @param {string} [releaseType] - release type (`patch`, `minor`, `major`, `prerelease`, `prepatch`, `preminor`, or `premajor`)
+* @throws {TypeError} must provide a string
+* @throws {Error} must provide a valid package name
+* @throws {TypeError} must provide a recognized release type
+* @returns {string} changelog contents
+*
+* @example
+* var changelog = generate( 'utils/curry' );
+* // returns '...'
+*
+* @example
+* var changelog = generate( 'stats/base/dists' );
+* // returns '...'
+*
+* @example
+* var changelog = generate( 'proxy' );
+* // returns '...'
+*
+* @example
+* var changelog = generate( 'utils/curry', 'patch' );
+* // returns '...'
+*
+* @example
+* var changelog = generate( 'utils/curry', 'major' );
+* // returns '...'
+*/
+function generate( pkg, releaseType ) {
+	var isNamespacePkg;
+	var releaseCommits;
+	var newestRelease;
+	var bySubpackage;
+	var nextVersion;
+	var standalone;
+	var unreleased;
+	var releases;
+	var pkgNames;
+	var commits;
+	var version;
+	var summary;
+	var date;
+	var name;
+	var out;
+	var i;
+	var j;
+
+	if ( !isString( pkg ) ) {
+		throw new TypeError( format( 'invalid argument. Must provide a string. Value: `%s`.', pkg ) );
+	}
+	if ( pkg === '' ) {
+		// Case: root package
+		isNamespacePkg = true;
+		standalone = '@stdlib/stdlib';
+		releases = objectEntries( npmReleases( standalone ) );
+	} else {
+		// Case: all other packages
+		isNamespacePkg = contains( STDLIB_NAMESPACE_PKGS, '@stdlib/' + pkg );
+		standalone = '@stdlib/' + replace( pkg, '/', '-' );
+		releases = objectEntries( npmReleases( standalone ) );
+	}
+
+	out = '# CHANGELOG\n\n';
+	out += '> Package changelog.\n\n';
+
+	commits = parseCommits( pkg );
```

Comment:
For this line, `parseCommits` does not need to be "package" aware. One can convert the package name to a directory path.